crux-core/src/crux/api/tx/MatchOperation.java

```diff
@@ -0,0 +1,71 @@
+package crux.api.tx;
+
+import crux.api.CruxDocument;
+
+import java.util.Date;
+import java.util.Objects;
+
+public final class MatchOperation extends TransactionOperation {
+    public static MatchOperation create(Object id) {
+        return new MatchOperation(id, null, null);
+    }
+
+    public static MatchOperation create(Object id, Date validTime) {
+        return new MatchOperation(id, null, validTime);
+    }
+
+    public static MatchOperation create(CruxDocument document) {
+        return new MatchOperation(document.getId(), document, null);
+    }
+
+    public static MatchOperation create(CruxDocument document, Date validTime) {
+        return new MatchOperation(document.getId(), document, validTime);
+    }
+
+    public Object getId() {
+        return id;
+    }
+
+    public CruxDocument getCompare() {
```

Comment:
`compare` isn't a word we use in this context elsewhere in Crux, so not sure I'd introduce it here. `document`?

Reply:
Yup, happy to change this